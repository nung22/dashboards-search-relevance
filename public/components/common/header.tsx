/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { FormattedMessage } from '@osd/i18n/react';
import { EuiText, EuiLink, EuiPanel, EuiTitle } from '@elastic/eui';

interface HeaderProps {
  children?: React.ReactNode;
}

export const Header = ({ children }: HeaderProps) => {
  return (
    <EuiPanel
      hasBorder={false}
      hasShadow={false}
      color="transparent"
      grow={false}
      borderRadius="none"
    >
      <EuiTitle>
        <h1>
          <FormattedMessage
            id="searchRelevance.header.title"
            defaultMessage="Compare search results"
          />
        </h1>
      </EuiTitle>
      <EuiText>
        <p>
          <FormattedMessage
            id="searchRelevance.header.description"
            defaultMessage="Compare results using the same search text with different queries. {learnMoreLink}"
            values={{
              learnMoreLink: (
                <EuiLink
                  href="https://opensearch.org/docs/latest/search-plugins/search-relevance"
                  target="_blank"
                >
                  <FormattedMessage
                    id="searchRelevance.header.description.learnMoreLinkLabel"
                    defaultMessage="Learn more"
                  />
                </EuiLink>
              ),
            }}
          />
        </p>
      </EuiText>
      {children}
    </EuiPanel>
  );
};
