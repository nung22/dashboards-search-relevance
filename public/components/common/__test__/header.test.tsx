/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { Header } from '../header';
import { IntlProvider } from 'react-intl';

describe('Header component', () => {
  configure({ adapter: new Adapter() });

  it('Renders header component', async () => {
    const wrapper = mount(
      <IntlProvider locale="en">
        <Header />
      </IntlProvider>
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
