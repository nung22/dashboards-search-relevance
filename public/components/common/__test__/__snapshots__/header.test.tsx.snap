// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Renders header component 1`] = `
<IntlProvider
  locale="en"
>
  <Header>
    <EuiPanel
      borderRadius="none"
      color="transparent"
      grow={false}
      hasBorder={false}
      hasShadow={false}
    >
      <div
        className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero"
      >
        <EuiTitle>
          <h1
            className="euiTitle euiTitle--medium"
          >
            <FormattedMessage
              defaultMessage="Compare search results"
              id="searchRelevance.header.title"
              values={Object {}}
            >
              <span>
                Compare search results
              </span>
            </FormattedMessage>
          </h1>
        </EuiTitle>
        <EuiText>
          <div
            className="euiText euiText--medium"
          >
            <p>
              <FormattedMessage
                defaultMessage="Compare results using the same search text with different queries. {learnMoreLink}"
                id="searchRelevance.header.description"
                values={
                  Object {
                    "learnMoreLink": <EuiLink
                      href="https://opensearch.org/docs/latest/search-plugins/search-relevance"
                      target="_blank"
                    >
                      <FormattedMessage
                        defaultMessage="Learn more"
                        id="searchRelevance.header.description.learnMoreLinkLabel"
                        values={Object {}}
                      />
                    </EuiLink>,
                  }
                }
              >
                <span>
                  Compare results using the same search text with different queries. 
                  <EuiLink
                    href="https://opensearch.org/docs/latest/search-plugins/search-relevance"
                    target="_blank"
                  >
                    <a
                      className="euiLink euiLink--primary"
                      href="https://opensearch.org/docs/latest/search-plugins/search-relevance"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      <FormattedMessage
                        defaultMessage="Learn more"
                        id="searchRelevance.header.description.learnMoreLinkLabel"
                        values={Object {}}
                      >
                        <span>
                          Learn more
                        </span>
                      </FormattedMessage>
                      <EuiIcon
                        aria-label="External link"
                        className="euiLink__externalIcon"
                        size="s"
                        type="popout"
                      >
                        <EuiIconEmpty
                          aria-hidden={true}
                          aria-label="External link"
                          className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                          focusable="false"
                          role="img"
                          style={null}
                        >
                          <svg
                            aria-hidden={true}
                            aria-label="External link"
                            className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                            focusable="false"
                            height={16}
                            role="img"
                            style={null}
                            viewBox="0 0 16 16"
                            width={16}
                            xmlns="http://www.w3.org/2000/svg"
                          />
                        </EuiIconEmpty>
                      </EuiIcon>
                      <EuiScreenReaderOnly>
                        <span
                          className="euiScreenReaderOnly"
                        >
                          <EuiI18n
                            default="(opens in a new tab or window)"
                            token="euiLink.newTarget.screenReaderOnlyText"
                          >
                            (opens in a new tab or window)
                          </EuiI18n>
                        </span>
                      </EuiScreenReaderOnly>
                    </a>
                  </EuiLink>
                </span>
              </FormattedMessage>
            </p>
          </div>
        </EuiText>
      </div>
    </EuiPanel>
  </Header>
</IntlProvider>
`;
